GSOC18/hexCells_to_squareCell_interpolation.py

```diff
@@ -0,0 +1,162 @@
+    t1=datetime.datetime.now()
+    print 'Reading completed in: ',t1-t0,' sec\n'
+
+    #Creating the empty energy map (initialized with zeros)
+    #coef=np.zeros(resolution,dtype=dtype)
+
+    #Iterating over all the cells to get the bounds of the detector
+    print '>>> Calculating Bounds'
+    center_x=map(lambda c:c.center.x,cells_dict.values())
+    max_x=max(center_x)
+    min_x=min(center_x)
+    center_y=map(lambda c:c.center.y,cells_dict.values())
+    max_y=max(center_y)
+    min_y=min(center_y)
+    t2=datetime.datetime.now()
+    print 'Bounding completed in: ',t2-t1,' sec\n'
+
+    #Calculating the maximum length of any cells
+        #(will to used to specify search radius in KD tree)
+    print '>>> Calculating the search radius'
+    max_length_hex=max(map(
+                    lambda c: max([
+                    c.vertices.bounds[2]-c.vertices.bounds[0],
+                    c.vertices.bounds[3]-c.vertices.bounds[1]
+                    ]),cells_dict.values())
+                    )
+    max_length_sq=np.sqrt( ((max_x-min_x)/(resolution[0]-1))**2+
+                           ((max_y-min_y)/(resolution[1]-1))**2 )
+    #Any overlapping cells will be in this search radius
+    search_radius=(max_length_hex/2)+(max_length_sq/2)
+    t3=datetime.datetime.now()
+    print 'Search Radius finding completed in: ',t3-t2,' sec\n'
+
+    #Getting the square cells mesh (dict) for overlap calculation
+    print '>>> Generating the square mesh grid'
+    sq_cells_dict=get_square_cells(resolution,min_x,min_y,max_x,max_y)
+    t4=datetime.datetime.now()
+    print 'Generating Mesh Grid completed in: ',t4-t3,' sec\n'
+
+    #Calculating the coefficient of overlap
+    #(currently in for of ditionary)
+    print '>>> Calculating the overlap coefficient'
+    coef=calculate_overlap(cells_dict.values(),sq_cells_dict.values(),
+                            search_radius,min_overlap_area=0.0)
+    t5=datetime.datetime.now()
+    print 'Overlap Coef Finding completed in: ',t5-t4,' sec\n'
+
+    #Now change it if we want the overlap with sq cells
+    #in form of array
+    #print coef
+    return coef
+
+
+def get_square_cells(resolution,min_x,min_y,max_x,max_y):
+    ''' This function will generate square mesh grid by Creating
+    the square polygon
+
+    '''
+    #Finding the dimension of each cells
+    x_length=(max_x-min_x)/(resolution[0]-1)
+    y_length=(max_y-min_y)/(resolution[1]-1)
+
+    #Creating empty array to store
+    #sq_cells=np.empty(resolution,dtype=np.object)
+    sq_cells={}
+
+    #Time Comlexity = O(res[0]*res[1])
+    for i in range(resolution[0]):
+        for j in range(resolution[1]):
+            #Center of the square polygon
+            center=(min_x+i*x_length,min_y+j*y_length)
+            id=(i,j)    #given in usual matrix notation
+            sq_cells[id]=sq_Cells(id,center,x_length,y_length)
+
+    #Saving the sq_cell sq_cell_data in given folder
+    fhandle=open(sq_cells_filename,'wb')
```

Comment:
The pickle file is being saved here for the square cells which I mentioned in the mail, according to filename given in "global variable" section of this file.